draft-ietf-jsonpath-base.md

```diff
@@ -1521,19 +1521,21 @@ A function expression is well-typed if all of the following are true:
   to have a result type of `LogicalType`, `SingleNodeType`, or `NodesType`.
 * If it occurs directly as a `comparable` in a comparison, the
   function is declared to have a result type of `ValueType` or `SingleNodeType`.
-* Otherwise, it occurs as an argument to a further function
-  expression and the following rules for function arguments apply to
-  its declared result type.
 * Each argument of the function conforms to the declared type of the corresponding declared
-  parameter according to one of the following rules:
-   * The argument is a function expression with declared result type that is the same as
-     the declared type of the parameter.
-   * The argument is a function expression with declared result type `SingleNodeType` and
-     the declared type of the parameter is `NodesType`.
-   * The argument is a literal primitive value and the declared type of the parameter is `ValueType`.
-   * The argument is a Singular Path and the declared type of the parameter is `SingleNodeType`.
-   * The argument is a Singular Path or `filter-path` (which includes
-     Singular Paths) and the declared type of the parameter is `NodesType`.
+  parameter according to one of the following scenarios:
+   * The declared type of the parameter is `ValueType` and the argument is one of:
+     * a function with declared type of `ValueType` or `SingleNodeType`
+     * a `singular-path`
```

Comment:
The PR needs to explain how to get from `SingleNodeType` and `singular-path` to `ValueType`. C.f. the following text from the "Comparisons" section:
> When any path on either side of a comparison results in a nodelist consisting of a single node, each such path is replaced by the value of its node

Reply:
The text on this branch actually reads:

> When any path **or function expression** on either side of a comparison results in a nodelist consisting of a single node, each such path **or function expression** is replaced by the value of its node and then...

so it's already there.

Reply:
Certainly that text is the kind of thing that is needed. But that text limits its application to either side of a comparison and this excludes arguments passed as parameters of type `ValueType`.

Reply:
Addressed in [ba8c16b](https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/pull/418/commits/ba8c16bf51f0aa495c605ca16e46006a1dbb3340)

Reply:
Yes, that does the job nicely.